### PR TITLE
NT-USP - American Security - How I Learned to Stop Worrying and Love the Gun

### PR DESCRIPTION
**Pull request by ToasterBiome**
# Document the changes in your pull request

Adds the NT-USP, a ballistic weapon available to security members on the station. In an early attempt to utilize hardlight technology, Nanotrasen developed weapons that synthesize small pieces of hardlight as projectiles. Although the test was a failure, it did give insight into hardlight usage in the disabling of criminals.

ezgif-4-a7cee03f5e

Statistics:
Max Ammo: 10 (Disabler is 20)
Damage = 20 Stamina, 2 Burn (Disabler is 25 Stamina)
Light EMPs remove 50% of shots from power packs, Heavy EMPs remove 100% of them.

You can charge small power pack batteries in weapon rechargers.

Security Members now do not spawn with their disabler, instead they may make their way to the Armaments Dispenser to choose which weapon they would prefer; a disabler, or an NT-USP and an extra clip:
image

https://user-images.githubusercontent.com/5091394/200694631-f6fab692-2b0e-46d9-a8f7-3874025eae3e.mp4

You can purchase more NT-USPs in cargo for 2000, giving you 3 NT-USPs and 3 extra power packs.

Armory spawns with 2 disablers and 1 NT-USP and power pack.

# Spriting
Aseprite_hLyPpaVBbG

Gives rechargable magazines a nicer sprite
Aseprite_1rolrxHUlP

# Wiki Documentation

Introduces the NT-USP, a disabler alternative. 
Guide to Combat stuff:
Ammo: Energy (Small Powerpack)
Capacity: 10 Rounds
Damage & Effects: 2 Burn, 20 Stamina
How to Acquire: 
- Armament Dispenser in Security
- The Armory
- Ordered From Cargo
Faction Usage:
Nanotrasen
Notes:
An alternative to the Disabler. Shoots .22HL projectiles, which function like bullets instead of disabler shots. The small powerpacks included can be recharged in any weapon recharger like a standard Disabler. EMPs will drain shots from the powerpacks within.

Disablers also need changing from "Standard issue for most Security personnel" to "Armament Dispenser in Security"

The Armament Dispenser is a new machine located in every Security outfitting area that gives you the choice of a Disabler or an NT-USP (along with an extra clip). One per security member. 

# Changelog

:cl:  ToasterBiome, Ynot01
rscadd: Adds the NT-USP and makes it available for Security
rscadd: Adds a 2000 cost NT-USP Crate with 3 NT-USPs and 3 clips
imageadd: Adds NT-USP sprite, Armament Dispenser sprite, and adds new sprites for power packs
mapping: Adds Armament Dispensers to every in-rotation map
/:cl:

### CODE REVIEW

**Comment by Bibby0110** [comment]
dumb

**Comment by MajManatee** [comment]
Epic. Poggers even.

**Comment by Chubbygummibear** [comment]
stay strapped with that mf thang on you

**Comment by LazennG** [comment]
pretty based and ballisticpilled i think

**Comment by Marmio64** [comment]
I suggest making sure there wont only be one dispenser on each map, at least one in the locker room and maybe 1 in the arrivals checkpoint so this doesn't make it easier to gank newcoming officers, or give them a chance to just not get a disabler if the sole one on the map was destroyed.

**Review by cuackles — APPROVED** [review]
hot

**Comment by Simplehorror** [comment]
why would anyone pick this instead of the disabler it sucks

**Comment by TheRyeGuyWhoWillNowDie** [comment]
> why would anyone pick this instead of the disabler it sucks

You suck.

**Review by KillerOrcaCora — CHANGES_REQUESTED** [review]
Revoking art approval, there is multiple issues that need to be addressed across these sprites, I will have everything sorted and drawn out when I get home.

**Comment by ToasterBiome** [comment]
> Revoking art approval, there is multiple issues that need to be addressed across these sprites, I will have everything sorted and drawn out when I get home.

Sure, contact me on Discord, biome?#5794, and we can sit down and go over the sprites and see what works and what doesn't

**Comment by Simplehorror** [comment]
Nice gun 👍

**Comment by Simplehorror** [comment]
> > why would anyone pick this instead of the disabler it sucks
> 
> You suck.

nah i take it back good mechanics + good sprites = happy

**Comment by Marmio64** [comment]
Can this shoot through windows? If not i'm really debating the usefulness of this considering you'd be having to reload for extra ammo instead of just having a double clip with no reload (you also need to use up inventory space to stock extra ammo), cant shoot through windows, and you deal less stamina damage each shot. Can't even get the advantage of EMP resistance since you'll be losing just about all your ammo regardless from an EMP, and seems like a really bad version of a WT equipped with rubber bullets that you're using because you cant get one.

**Comment by ToasterBiome** [comment]
> Can this shoot through windows? If not i'm really debating the usefulness of this considering you'd be having to reload for extra ammo instead of just having a double clip with no reload (you also need to use up inventory space to stock extra ammo), cant shoot through windows, and you deal less stamina damage each shot. Can't even get the advantage of EMP resistance since you'll be losing just about all your ammo regardless from an EMP, and seems like a really bad version of a WT equipped with rubber bullets that you're using because you cant get one.

Nope. It's a gun. 

I'm up to ideas for balancing as risk v reward of it, right now it's just "disabler but worse so it will be merged"

**Comment by 00ze-cyclone** [comment]
> Can this shoot through windows? If not i'm really debating the usefulness of this considering you'd be having to reload for extra ammo instead of just having a double clip with no reload (you also need to use up inventory space to stock extra ammo), cant shoot through windows, and you deal less stamina damage each shot. Can't even get the advantage of EMP resistance since you'll be losing just about all your ammo regardless from an EMP, and seems like a really bad version of a WT equipped with rubber bullets that you're using because you cant get one.

Rn now biggest advantage is that it shoot bullets that will go twice as fast as a regular disabler beam so making shots hit is way easier and you can hoard mags though it’s not as simple as for other ballistic weapons.
Also smaller than disabler so you can fit it in pockets.

**Comment by ReddicusDragon** [comment]
based and america pilled. Is it possible to give officers disablers round start and make it so the dispenser requires you to give it a disabler before it gives you the USP instead? So on round join sec offs have some form of protection?

**Comment by Potototatoh** [comment]
could lock the disabler behind research

**Comment by MajManatee** [comment]
The fact that the mags aren't entirely emptied when EMP'd already makes this better than a disabler in my eyes. Sure, I have to carry some spare mags, but no more will literally everything I have on me become useless because a traitor spend 3TC on an EMP pack.

**Comment by ToasterBiome** [comment]
> The fact that the mags aren't entirely emptied when EMP'd already makes this better than a disabler in my eyes. Sure, I have to carry some spare mags, but no more will literally everything I have on me become useless because a traitor spend 3TC on an EMP pack.

Light EMPs now delete 50% of bullets, and Heavy EMP's delete 100% of the bullets, this keeps parity with the Disabler.

**Comment by KillerOrcaCora** [comment]
The Gun is better, but the small power pack could use a tiny size increase for the sake of the UI. The actual armaments machine could also be stretched a tiny bit so it's at least the same size as the sec-drobe but you can do whatever you want with that.

armaments dispenser is missing an emag effect so it can be accessed by people other than security or the warden. 

Last thing, Security and the warden lose the disablers that they would get from joining the shift late which leaves them more prone to threats. Should probably make some snow flake code for late joiners.

**Review by KillerOrcaCora — CHANGES_REQUESTED** [review]
oops

**Comment by ToasterBiome** [comment]
> The Gun is better, but the small power pack could use a tiny size increase for the sake of the UI. The actual armaments machine could also be stretched a tiny bit so it's at least the same size as the sec-drobe but you can do whatever you want with that.
> 
> armaments dispenser is missing an emag effect so it can be accessed by people other than security or the warden.
> 
> Last thing, Security and the warden lose the disablers that they would get from joining the shift late which leaves them more prone to threats. Should probably make some snow flake code for late joiners.

I can address all these issues shortly. Ynot and I had the idea of including one in the sec arrivals checkpoint and also an extra board in the armory. The emag effect is a good idea.

**Comment by ToasterBiome** [comment]
Waiting for a change to the armament dispenser, dnm for now

**Comment by kooarbiter** [comment]
cool idea, but by god will the burn damage be annoying to fix since only brigbay doctors have medical access, have fun carrying around surplus burn gel, either get rid of the burn damage (on a supposed non lethal weapon) or restrict it to code somewhere between blue and red

**Comment by JamieD1** [comment]
> > The Gun is better, but the small power pack could use a tiny size increase for the sake of the UI. The actual armaments machine could also be stretched a tiny bit so it's at least the same size as the sec-drobe but you can do whatever you want with that.
> > armaments dispenser is missing an emag effect so it can be accessed by people other than security or the warden.
> > Last thing, Security and the warden lose the disablers that they would get from joining the shift late which leaves them more prone to threats. Should probably make some snow flake code for late joiners.
> 
> I can address all these issues shortly. Ynot and I had the idea of including one in the sec arrivals checkpoint and also an extra board in the armory. The emag effect is a good idea.

There should be no disablers spawned in the map, there need to be a limited supply of them, Maybe make a machine which changes the disabler into your pistol and vice versa?

**Comment by ToasterBiome** [comment]
> > > The Gun is better, but the small power pack could use a tiny size increase for the sake of the UI. The actual armaments machine could also be stretched a tiny bit so it's at least the same size as the sec-drobe but you can do whatever you want with that.
> > > armaments dispenser is missing an emag effect so it can be accessed by people other than security or the warden.
> > > Last thing, Security and the warden lose the disablers that they would get from joining the shift late which leaves them more prone to threats. Should probably make some snow flake code for late joiners.
> > 
> > 
> > I can address all these issues shortly. Ynot and I had the idea of including one in the sec arrivals checkpoint and also an extra board in the armory. The emag effect is a good idea.
> 
> There should be no disablers spawned in the map, there need to be a limited supply of them, Maybe make a machine which changes the disabler into your pistol and vice versa?

I originally wanted to make it a pref, but ynot had the idea to make it a machine where you get 1 per ID, and I think that's a bit cooler than just "ok you spawn with your gun you want", problem is making them available everywhere.

**Comment by UltraBigRat** [comment]
how many clips can you get roundstart? since the main advantage here is being reloadable that really decides the viability of the weapon

**Comment by ToasterBiome** [comment]
> how many clips can you get roundstart? since the main advantage here is being reloadable that really decides the viability of the weapon

You get a clip in the gun and 1 extra clip

**Comment by JamieD1** [comment]
I would prefer a preference to be honest, or we could do the one per ID I guess, just need to make sure there are none that people can just hack it and get.

**Comment by Bibby0110** [comment]
I don’t see a reason for this to be merged unless they offer a distinct difference compared to regular disablers.

**Comment by Chubbygummibear** [comment]
where's my antag token

**Comment by ToasterBiome** [comment]
> where's my antag token

You're mama

**Comment by Burger1998** [comment]
> I don’t see a reason for this to be merged unless they offer a distinct difference compared to regular disablers.

bang bang cooler than pew pew

**Comment by 00ze-cyclone** [comment]
> I don’t see a reason for this to be merged unless they offer a distinct difference compared to regular disablers.

Basically reliability in exchange for power, easier to land hits because bullets are faster and the gun is less murdered by EMPs but less damage.

**Comment by chemistrymain2** [comment]
> I don’t see a reason for this to be merged unless they offer a distinct difference compared to regular disablers.

the distinct difference here is that it's a ballistic, sacrificing the ability to penetrate windows and being handled by bullet rather than energy armor in exchange for easier and more effective reloads

bullets are not faster (they travel at the same speed as other projectiles) @00ze-cyclone 

but you can effectively put out more of these, and they will be handled by ballistic armor instead of energy armor, so people with the ablative are less resilient and bulletproof armor is more resilient

basically? same reason people would use riot foam darts-surprisingly useful for volume of fire and easier reloads but sacrificing window spam and the like

**Comment by 00ze-cyclone** [comment]
Bullets are actually faster than energy projectiles, they have been made twice as fast as lasers and such (bullet got 0.4 so 0.4 second to travel through a tile while energy projectiles remained at 0.8)

**Comment by MajManatee** [comment]
hurry up and fix this so i can merge it smh

**Comment by Firewolf34** [comment]
Love the concept, but I am against a pref for this. If you make this a preference then unless you've *perfectly* balanced this, there will just be some optimal meta and everyone will micro-optimize and choose one or the other. 

Whereas, if it's some kinda unlock, players will still get to experience both weapons in some quantity. I think it's important to avoid that sort of meta because if one of the weapons has any sort of edge, the other will simply be a noob trap. It's not a fun game design pattern.

I like how you've got it with the machine. Seems like a good move to allow for future weapons too.

**Comment by JamieD1** [comment]
@ToasterBiome fix the conflict then we can merge

**Review by JamieD1 — APPROVED** [review]

**Review by KillerOrcaCora — APPROVED** [review]
